packages/sit-onyx/src/utils/forms.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Extracts the first invalid validity type from the given HTML ValidityState.
+ * "valueMissing" is prioritized over other types to align with the default browser behavior.
+ */
+export const getFirstInvalidType = (validity: ValidityState) => {
+  // since the types are getters of the ValidityState we need to get the keys using "Object.getOwnPropertyDescriptors"
+  const availableValidityTypes = Object.entries(
+    Object.getOwnPropertyDescriptors(ValidityState.prototype),
+  )
+    .filter(([key, value]) => key !== "valid" && value.enumerable)
```

Comment:
```suggestion
    .filter(([key, value]) => key !== "valid")
```

isn't this redundant?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties?retiredLocale=de
> Most iteration means (such as [for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loops and [Object.keys](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)) only visit enumerable keys.

Reply:
same for Object.entries
> The Object.entries() static method returns an array of a given object's own **enumerable** string-keyed property key-value pairs.

Reply:
Does not seem so because keys like `constructor` are still included
![image](https://github.com/SchwarzIT/onyx/assets/67898185/0c4a4e11-4b1d-40c5-bb8f-e40821496dd8)